storage/database/sharded_database_test.go

```diff
@@ -0,0 +1,304 @@
+// Copyright 2021 The klaytn Authors
+// This file is part of the klaytn library.
+//
+// The klaytn library is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Lesser General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// The klaytn library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public License
+// along with the klaytn library. If not, see <http://www.gnu.org/licenses/>.
+package database
+
+import (
+	"bytes"
+	"context"
+	"io/ioutil"
+	"os"
+	"reflect"
+	"sort"
+	"sync"
+	"testing"
+
+	"github.com/klaytn/klaytn/common"
+
+	"github.com/stretchr/testify/assert"
+)
+
+var ShardedDBConfig = []*DBConfig{
+	{DBType: LevelDB, SingleDB: false, NumStateTrieShards: 4, ParallelDBWrite: true},
+	{DBType: LevelDB, SingleDB: false, NumStateTrieShards: 8, ParallelDBWrite: true},
+	{DBType: LevelDB, SingleDB: false, NumStateTrieShards: 16, ParallelDBWrite: true},
+}
+
+// testIterator tests if given iterator iterates all entries
+func testIterator(t *testing.T, checkOrder bool, entryNums []uint, dbConfig []*DBConfig, entriesFromIterator func(t *testing.T, db shardedDB, entryNum uint) []common.Entry) {
+	for _, entryNum := range entryNums {
+		entries := common.CreateEntries(int(entryNum))
+		dbs := make([]shardedDB, len(dbConfig))
+
+		// create DB and write data for testing
+		for i, config := range dbConfig {
+			config.Dir, _ = ioutil.TempDir(os.TempDir(), "test-shardedDB-iterator")
+			defer func(dir string) {
+				if err := os.RemoveAll(dir); err != nil {
+					t.Fatalf("fail to delete file %v", err)
+				}
+			}(config.Dir)
+
+			// create sharded DB
+			db, err := newShardedDB(config, 0, config.NumStateTrieShards)
+			dbs[i] = *db
+			if err != nil {
+				t.Log("Error occured while creating DB")
+				t.FailNow()
+			}
+
+			// write entries data in DB
+			batch := db.NewBatch()
+			for _, entry := range entries {
+				assert.NoError(t, batch.Put(entry.Key, entry.Val))
+			}
+			assert.NoError(t, batch.Write())
+		}
+
+		// sort entries for each compare
+		sort.Slice(entries, func(i, j int) bool { return bytes.Compare(entries[i].Key, entries[j].Key) < 0 })
+
+		// get data from iterator and compare
+		for _, db := range dbs {
+			// create iterator
+			entriesFromIt := entriesFromIterator(t, db, entryNum)
+			if !checkOrder {
```

Comment:
Why does this sort if `checkOrder` is false?

Reply:
When the order is not checked, I just wanted to check if every entries are iterated. It is sorted to pass `DeepEqual` later in the code.

Reply:
Got it. :)